Bottom: 1dcd1d0ebf4fc29d6eeff100d0e6f3f786396996
Top:    4cc67c45a1f87dcea7bf98526bed8a70433c7ee6
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-07 14:54:50 +0800

autofs-5.0.7 - add initialization of bind_result in-do_sasl_bind().patch

There is an unlikley code path where bind_result could be used uninitialized
so initialize it so it isn't incorrectly used if it has rubish in it.


---

diff --git a/modules/cyrus-sasl.c b/modules/cyrus-sasl.c
index b456333..68f9242 100644
--- a/modules/cyrus-sasl.c
+++ b/modules/cyrus-sasl.c
@@ -210,7 +210,7 @@ int
 do_sasl_bind(unsigned logopt, LDAP *ld, sasl_conn_t *conn, const char **clientout,
 	     unsigned int *clientoutlen, const char *auth_mech, int sasl_result)
 {
-	int ret, msgid, bind_result;
+	int ret, msgid, bind_result = LDAP_OTHER;
 	struct berval client_cred, *server_cred, temp_cred;
 	LDAPMessage *results;
 	int have_data, expected_data;
